### ISSUE TITLE
"Could not detect an installed debugger compatible with this file"

### ISSUE THREAD

**Issue by chris-revell** [31i73/atom-dbg#45]
Error message received whilst trying to debug `.cpp` file on a Mac. GDB installed at `/usr/local/bin/gdb`. Any suggestions?

**Comment by my-vegetable-has-exploded** [comment]
the same problem

**Comment by cccnrc** [comment]
same problem

**Comment by Lulz322** [comment]
Try to debug a.out file :)

**Comment by Moenish** [comment]
Same problem here but on windows...

Edit: So, in the description it says that it debugs executable files. It won't work with cpp files unless you compile them into a .exe file and then debug that file (well, it worked for me like this on windows, but I don't know how to do it on a Mac)

**Comment by penpendandan** [comment]
same problem，how to fix it

**Comment by Lulz322** [comment]
Guys, need to debug Executable File(File after gcc/g++), NOT a .CPP files.

**Comment by chris-revell** [comment]
> Guys, need to debug Executable File(File after gcc/g++), NOT a .CPP files.

This is actually what I was doing - my initial explanation was poorly worded. I never did find a solution.

**Comment by Lulz322** [comment]
> 
> 
> > Guys, need to debug Executable File(File after gcc/g++), NOT a .CPP files.
> 
> This is actually what I was doing - my initial explanation was poorly worded. I never did find a solution.

I have the same text trouble that in a topic because i tried to debug .cpp file, when need to debug execution file. In your first message u said " trying to debug .cpp file" :)

**Comment by Lulz322** [comment]
> 
> 
> > Guys, need to debug Executable File(File after gcc/g++), NOT a .CPP files.
> 
> This is actually what I was doing - my initial explanation was poorly worded. I never did find a solution.

Ohhh, u have macOS, ok, try to install lldb
lldb better working on MacOS then gdb.

**Comment by chris-revell** [comment]
> > > Guys, need to debug Executable File(File after gcc/g++), NOT a .CPP files.
> > 
> > 
> > This is actually what I was doing - my initial explanation was poorly worded. I never did find a solution.
> 
> I have the same text trouble that in a topic because i tried to debug .cpp file, when need to debug execution file. In your first message u said " trying to debug .cpp file" :)

Yes that was my mistake. As for lldb, I use this in the terminal but have yet to find an atom package for it unfortunately.
